### PR TITLE
Get Pdf for letter

### PR DESCRIPTION
**Pull request by leohemsted**
## Checklist

- [x] I’ve used the pull request template
- [x] I’ve written unit tests for these changes
- [x] I’ve update the documentation (in `DOCUMENATION.md` and `CHANGELOG.md`)
- [x] I’ve bumped the version number (in `src/Notify/Notify.csproj`)

### CODE REVIEW

File: src/Notify.Tests/IntegrationTests/NotificationClientIntegrationTests.cs [hunk 1]
```diff
@@ -476,6 +479,26 @@ public void SendSmsTestWithPersonalisationAndSmsSenderId()
...
+					if (!e.Message.Contains("PDFNotReadyError")) {
+						throw e;
+					} else {
+						Thread.Sleep(3000);
+					}
+				}
+			}
+			Assert.IsNotNull(pdfData);
+			var expectedResponse = Encoding.UTF8.GetBytes("%PDF-");
+			Assert.AreEqual(pdfData.Take(expectedResponse.Length).ToArray(), expectedResponse);
```
[the remark this answers is not in the document]
Comment (leohemsted) [reply src/Notify.Tests/IntegrationTests/NotificationClientIntegrationTests.cs]: the PDF is generated by notify, so we don't know anything about what it contains. All we can check here is that it is a pdf file.

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -777,6 +777,44 @@ If the request is not successful, the client returns a `Notify.Exceptions.Notify
...
+
+#### notificationId (required)
+
+The ID of the notification. You can find the notification ID in the response to the [original notification method call](#get-the-status-of-one-message-response).
+
+You can also find it by signing in to [GOV.UK Notify](https://www.notifications.service.gov.uk) and going to the __API integration__ page.
+
+### Response
+
+If the request to the client is successful, the client will return a `bytes` object containing the raw PDF data.
```
[the remark this answers is not in the document]
Comment (leohemsted) [reply DOCUMENTATION.md]: `byte[]`. good spot

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -777,6 +777,44 @@ If the request is not successful, the client returns a `Notify.Exceptions.Notify
...
+
+|error.status_code|error.message|How to fix|
+|:---|:---|:---|
+|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "id is not a valid UUID"`<br>`}]`|Check the notification ID|
+|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "Notification is not a letter"`<br>`}]`|Check that you are looking up the correct notification|
+|`400`|`[{`<br>`"error": "PDFNotReadyError",`<br>`"message": "PDF not available yet, try again later"`<br>`}]`|Wait for the notification to finish processing. This usually takes a few seconds|
+|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Document did not pass the virus scan"`<br>`}]`|You cannot retrieve the contents of a letter notification that contains a virus|
+|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "PDF not available for letters in technical-failure"`<br>`}]`|You cannot retrieve the contents of a letter notification in technical-failure|
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Error: Your system clock must be accurate to within 30 seconds"`<br>`}]`|Check your system clock|
```
Comment (idavidmcdonald) [inline DOCUMENTATION.md]: Optional: I don't think it's worth adding the 403 messages here to do with auth. They feel too generic.

Comment (leohemsted) [reply DOCUMENTATION.md]: i think consistency is more important. we've followed this patterns elsewhere, both in this file and in the other languages. i think.

**Pull request merged**